freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaInstanceHealthDetailsService.java

```diff
@@ -106,18 +114,58 @@ public NodeHealthDetails getInstanceHealthDetails(Stack stack, InstanceMetaData
     }
 
     private RPCResponse<Boolean> toBooleanRpcResponse(RPCResponse<CheckResult> nodeHealth) {
+        RPCMessage rpcMessage = createRpcMessageFromCheckResult(nodeHealth);
         RPCResponse<Boolean> response = new RPCResponse<>();
         response.setSummary(nodeHealth.getSummary());
         response.setResult(isHealthCheckPassing(nodeHealth));
         response.setCount(nodeHealth.getCount());
         response.setTruncated(nodeHealth.getTruncated());
-        response.setMessages(nodeHealth.getMessages());
+        response.setMessages(List.of(rpcMessage));
         response.setCompleted(nodeHealth.getCompleted());
         response.setFailed(nodeHealth.getFailed());
         response.setValue(nodeHealth.getValue());
         return response;
     }
 
+    private RPCMessage createRpcMessageFromCheckResult(RPCResponse<CheckResult> nodeHealth) {
+        List<CheckEntry> checkEntries = filterCheckEntries(nodeHealth);
+        List<PluginStatusEntry> pluginStatusEntries = filterPluginStatusEntries(nodeHealth);
+        CheckResult filteredCheckResult = new CheckResult();
+        filteredCheckResult.setChecks(checkEntries);
+        filteredCheckResult.setPluginStats(pluginStatusEntries);
+        if (nodeHealth.getResult() != null) {
+            filteredCheckResult.setStatus(nodeHealth.getResult().getStatus());
+            filteredCheckResult.setHost(nodeHealth.getResult().getHost());
+        }
+        RPCMessage rpcMessage = new RPCMessage();
+        rpcMessage.setName("node health check");
+        rpcMessage.setMessage(JsonUtil.writeValueAsStringSilent(filteredCheckResult));
+        if (nodeHealth.getFirstRpcMessage() != null) {
+            rpcMessage.setCode(nodeHealth.getFirstRpcMessage().getCode());
+        }
+        return rpcMessage;
+    }
+
+    private List<PluginStatusEntry> filterPluginStatusEntries(RPCResponse<CheckResult> nodeHealth) {
+        if (nodeHealth.getResult() == null || nodeHealth.getResult().getPluginStats() == null) {
+            return List.of();
+        } else {
+            return nodeHealth.getResult().getPluginStats().stream()
+                    .filter(not(check -> HEALTHY.equals(check.getStatus())))
+                    .collect(Collectors.toList());
+        }
+    }
+
+    private List<CheckEntry> filterCheckEntries(RPCResponse<CheckResult> nodeHealth) {
+        if (nodeHealth.getResult() == null || nodeHealth.getResult().getChecks() == null) {
+            return List.of();
+        } else {
+            return nodeHealth.getResult().getChecks().stream()
+                    .filter(not(check -> HEALTHY.equals(check.getStatus())))
```

Comment:
Does this change the behavior of how "DEGRADED" health status is reported? I think we are OK but I just wanted to see if that is something that was thought about. I see some tests with unhealthy but not degraded.

I think when the degraded status is reported from the thunderhead freeipa health agent, the API is supposed to return a 200 so that things like cluster proxy are OK. How I forget how this ends up being reported in the freeipa management service API.

Reply:
the only thing is changed, what's in the `RpcResponse#message` field: https://github.com/hortonworks/cloudbreak/pull/12142/files#diff-2b6708e2123104b8f8e4ec3515e4b80e4ab8231962296d9378446a72c1a6f7e4R123
Which is used here to construct the message (statusreason would get the value from here): https://github.com/hortonworks/cloudbreak/blob/31b2625a0956ee11d232b44c851d0a6438358450/freeipa/src/main/java/com/sequenceiq/freeipa/sync/FreeipaChecker.java#L184
 I haven't found any other usage of this field going up the call hierarchy of the changed method. Also status is checked/calculated from `RpcResponse#status` field, which is not changed here